routes/database.js

```diff
@@ -8,480 +8,382 @@ let my_conf = cfgsrv.get_conf();
-        session_user = await dbsrv.mongo_users().findOne({_id: req.locals.logInfo.id});
-        isadmin = await rolsrv.is_admin(session_user);
+        session_user = await dbsrv.mongo_users().findOne({ _id: req.locals.logInfo.id });
     } catch(e) {
         logger.error(e);
-        res.status(404).send({message: 'User session not found'});
-        res.end();
+        res.status(404).send({ message: 'User session not found' });
         return;
     }
-
-    if (!session_user){
-        res.status(401).send({message: 'Not authorized'});
+    if (!session_user) {
+        res.status(401).send({ message: 'Not authorized' });
         return;
     }
 
-    session_user.is_admin = isadmin;
-
-    let filter = {owner: req.params.owner};
-    let databases = await dbsrv.mongo_databases().find(filter).toArray();
+    let databases = await dbsrv.mongo_databases().find({ owner: req.params.owner }).toArray();
     res.send(databases);
-    return;
 });
 
-router.post('/database/:id', async function(req, res) {
-    if(! req.locals.logInfo.is_logged) {
-        res.status(401).send({message: 'Not authorized'});
+
+router.post('/requestdatabase/:id', async function(req, res) {
+    if (!req.locals.logInfo.is_logged) {
+        res.status(401).send({ message: 'Not authorized' });
         return;
     }
-    if(! sansrv.sanitizeAll([req.params.id])) {
-        res.status(403).send({message: 'Invalid parameters'});
+    if (!sansrv.sanitizeAll([req.params.id])) {
+        res.status(403).send({ message: 'Invalid parameters' });
         return;
     }
-    let session_user = null;
-    let isadmin = false;
+    let session_user;
+    let is_admin;
     try {
-        session_user = await dbsrv.mongo_users().findOne({_id: req.locals.logInfo.id});
-        isadmin = await rolsrv.is_admin(session_user);
-    } catch(e) {
+        session_user = await dbsrv.mongo_users().findOne({ _id: req.locals.logInfo.id });
+        is_admin = await rolsrv.is_admin(session_user);
+    } catch (e) {
         logger.error(e);
-        res.status(404).send({message: 'User session not found'});
-        res.end();
+        res.status(404).send({ message: 'User session not found' });
         return;
     }
-
     if (!session_user) {
-        res.status(401).send({message: 'Not authorized'});
+        res.status(401).send({ message: 'Not authorized' });
+        return;
+    }
+    if (req.body.owner && req.body.owner !== session_user.uid && !is_admin) {
+        res.status(401).send({ message: 'Not authorized, cannot request a database for a different user' });
+        return;
+    }
+    if (!req.body.expire) {
+        res.status(403).send({ message: 'No expiration date' });
         return;
     }
+    let pending_db = {
+        owner: req.body.owner ? req.body.owner : session_user.uid,
+        name: req.params.id,
+        type: req.body.type ? req.body.type : 'mysql',
+        host: req.body.host && sansrv.sanitize(req.body.host) ? req.body.host : CONFIG.mysql.host,
```

Comment:
Should be restricted to the default values I think. We don't want users requesting something other than mysql & the config.host